octopus-samples-instances/azure-base-infrastructure-terraform/azure-mysql.tf

```diff
@@ -1,37 +1,37 @@
-resource "azurerm_mysql_server" "permanent" {
-    name                             = var.azure_mysql_name
-    resource_group_name              = azurerm_resource_group.permanent.name
-    location                         = azurerm_resource_group.permanent.location
-    
-    administrator_login              = var.azure_mysql_administrator_name
-    administrator_login_password     = var.azure_mysql_administrator_password
+resource "azurerm_mysql_flexible_server" "permanent" {
+  name                = var.azure_mysql_name
+  resource_group_name = azurerm_resource_group.permanent.name
+  location            = azurerm_resource_group.permanent.location
+
+  administrator_login    = var.azure_mysql_administrator_name
+  administrator_password = var.azure_mysql_administrator_password
+
+  sku_name                     = "B_Standard_B1ms"
+  version                      = "8.0"
+  backup_retention_days        = 7
+  geo_redundant_backup_enabled = false
 
-    sku_name                         = "B_Gen5_2"
-    version                          = "8.0"
-    storage_mb                       = 5120
-    backup_retention_days            = 7
-    geo_redundant_backup_enabled     = false
-    auto_grow_enabled                = false
-    public_network_access_enabled    = true
-    ssl_enforcement_enabled          = true
-    ssl_minimal_tls_version_enforced = "TLS1_2"
+  storage {
+    auto_grow_enabled = false
+    size_gb           = 20

```

Comment:
QQ: is 20Gb the minimum, or can we specify anything lower?

Reply:
Yeah, that is a weird one - the [docs for the resource](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/mysql_flexible_server) specify 20 as the minimum, but I'll try and true that up with the Azure portal and see what I can find out

Reply:
Ta

Reply:
I read the docs properly this time. That option is the _maximum_ amount of storage, not minimum so you're probably all good with this, and no need to test:

> [size_gb](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/mysql_flexible_server#size_gb) - (Optional) The max storage allowed for the MySQL Flexible Server. Possible values are between 20 and 16384.

Reply:
I'd go with a lower backup retention.  In theory, no backups should be occurring, but if there are, that would save space ... unless that's the lowest you can go.

Reply:
Minimum backup retention is 1 day according to the TF docs

Reply:
Lowered to 1 day for MySQL in the last commit - Postgres min is still 7 days, so leaving as is for now 👍